yapapi/executor/agreements_pool.py

```diff
@@ -0,0 +1,118 @@
+import asyncio
+from dataclasses import dataclass
+import datetime
+import random
+from typing import Dict, NamedTuple, Optional
+from yapapi.executor import events
+from yapapi.props import NodeInfo
+from yapapi.rest.market import Agreement, OfferProposal
+
+
+class _BufferedProposal(NamedTuple):
+    ts: datetime.datetime
+    score: float
+    proposal: OfferProposal
+
+
+@dataclass
+class BufferedAgreement:
+    agreement: Agreement
+    worker_task: Optional[asyncio.Task]
+    has_multi_activity: bool
+
+
+class AgreementsPool:
+    """Manages proposals and agreements pool"""
+    _offer_buffer:Dict[str, _BufferedProposal] = {}  # provider_id -> Proposal
+    _agreements:Dict[str, BufferedAgreement] = {}  # agreement_id -> Agreement
+    _lock = asyncio.Lock()
+    confirmed = 0
+
+    def __init__(self, emitter):
+        self.emitter = emitter
+
+    async def cycle(self):
```

Comment:
Can you add short comments (in the form of function docstrings) explaining what each method does?